PowerValidatedSolutions.psm1

```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
+
+    #>
+
+    [CmdletBinding(DefaultParametersetName = "credentials")][OutputType('System.Management.Automation.PSObject')]
+
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+    )
+
+    Try {
+        $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/nsxt-managers"
+        if ($PSBoundParameters.ContainsKey('certificate') -and $PSBoundParameters.ContainsKey('privatekey')) {
+              if (Test-Path -Path $certificate) {
+                $cert = (Get-Content -Path $certificate) -join "`n"
+              }
+              else {Write-Error "Unable to find certificate file ($certificate)"}
+              if (Test-Path -Path $privatekey) {
+                $key = (Get-Content -Path $privatekey) -join "`n"
+              }
+              else {Write-Error "Unable to find key file ($privatekey)"}
+            $body = @{
+                fqdn               = $fqdn
+                proxy_id           = $collectorId
+                nickname           = $nickname
+                enabled            = $enabled
+                ipfix_enabled      = $true
+                latency_enabled    = $true
+                cred_type          = "CERTIFICATE"
+                client_certificate = $cert
+                client_private_key = $key
+              } | ConvertTo-Json
+        } elseif ($PSBoundParameters.ContainsKey('username') -and $PSBoundParameters.ContainsKey('password')) {
+            $body = @{
+                fqdn            = $fqdn
+                proxy_id        = $collectorId
+                nickname        = $nickname
+                enabled         = $enabled
+                ipfix_enabled   = $true
+                latency_enabled = $true
+                credentials     = @{
+                  username = $username
+                  password = $password
+                }
+              } | ConvertTo-Json
+          }
+            Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
+  } Catch {
+        Write-Error $_.Exception.Message
+  }
+}
```

Comment:
```suggestion
    Try {
        $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/nsxt-managers"
        if ($PSBoundParameters.ContainsKey('certificate') -and $PSBoundParameters.ContainsKey('privatekey')) {
            if (Test-Path -Path $certificate) {
                $cert = (Get-Content -Path $certificate) -join "`n"
            } else {
                Write-Error "Unable to find certificate file ($certificate)"
            }
            if (Test-Path -Path $privatekey) {
                $key = (Get-Content -Path $privatekey) -join "`n"
            } else {
                Write-Error "Unable to find key file ($privatekey)"
            }
            $body = @{
                fqdn = $fqdn
                proxy_id = $collectorId
                nickname = $nickname
                enabled = $enabled
                ipfix_enabled = $true
                latency_enabled = $true
                cred_type = "CERTIFICATE"
                client_certificate = $cert
                client_private_key = $key
            } | ConvertTo-Json
        } elseif ($PSBoundParameters.ContainsKey('username') -and $PSBoundParameters.ContainsKey('password')) {
            $body = @{
                fqdn = $fqdn
                proxy_id = $collectorId
                nickname = $nickname
                enabled = $enabled
                ipfix_enabled = $true
                latency_enabled = $true
                credentials = @{
                    username = $username
                    password = $password
                }
            } | ConvertTo-Json
        }
        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
    } Catch {
        Write-Error $_.Exception.Message
    }
}
```

Reply:
Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80